src/app_service/service/shared_urls/dto/url_data.nim

```diff
@@ -56,12 +61,26 @@ proc toContactUrlDataDto*(jsonObj: JsonNode): ContactUrlDataDto =
   discard jsonObj.getProp("description", result.description)
   discard jsonObj.getProp("publicKey", result.publicKey)
 
+proc toShardUrlDataDto*(jsonObj: JsonNode): ShardUrlDataDto =
+  result = ShardUrlDataDto()
+  discard jsonObj.getProp("cluster", result.cluster)
+  discard jsonObj.getProp("index", result.index)
+
 proc toUrlDataDto*(jsonObj: JsonNode): UrlDataDto =
   result = UrlDataDto()
 
   var communityObj: JsonNode
   if (jsonObj.getProp("community", communityObj)):
     result.community = communityObj.toCommunityUrlDataDto()
+
+  var shardObj: JsonNode
+  if (jsonObj.getProp("shard", shardObj)):
+    result.shard = shardObj.toShardUrlDataDto()
+  else:
+    result.shard = ShardUrlDataDto(
```

Comment:
I tried, but Nim doesn't let me. We can't pass `nil` to something that has a type. That might just be missing knowledge from my part?

Reply:
Hmm, I don't get it. It can be a pointer to `Shard`, nope?

Reply:
You mean doing this?
```nim
else:
    result.shard = Shard()
```
If so, I prefer setting the values to -1, as that way we can easily check the values before sending to status-go. I think they would default to 0 otherwise and those can exist I think

Reply:
I mean that if we change the definition of `Shard` to be a `ref object`:
```nim
type Shard* =  ref object
```

... or make the particular field to be `ref`, but this will require more changes, I'd go with first way.
```nim
  shard*: ref Shard
```

Then u can with no problems set `result.shard = nil`. Tested, compiles 😄

Reply:
Updated @igor-sirotin you can review again